### ISSUE TITLE
'500 Internal Server Error' recieved when POSTing to function that has environment variables set in the YAML

### ISSUE THREAD

**Issue by ericstoekl** [openfaas/faas#213]
I was trying out the `environment:` YAML element to set environment variables in the deployment file. However, it causes some problems for me. When I deploy the following function:

```
provider:
  name: faas
  gateway: http://localhost:8080

functions:
  gif-maker:
    lang: Dockerfile
    handler: ./gif-maker
    image: ems5311/gif-maker:latest
    environment:
      read_timeout: 65
      write_timeout: 65
      write_debug: false
```

When I go to use this function (`cat file.mp4 | faas-cli invoke -f file.yml`) I always get a `500 Internal Server Error` response from the function. When I remove the environment variables `read_timeout: 65` and `write_timeout: 65`, the function suddenly accepts requests and works just fine.

Strangely, I have another function deployed with `read_timeout:` and `write_timeout:` env variables set in the YAML, and it works fine.

Is this feature just not implemented yet?

Thanks,
Eric

## Steps to Reproduce (for bugs)

1.
2.
3.
4.

## Your Environment

* Docker version `docker version` (e.g. Docker 17.0.05 ):
**17.09.0-ce-rc1**
* Are you using Docker Swarm or Kubernetes (FaaS-netes)?
**Docker Swarm**
* Operating System and version (e.g. Linux, Windows, MacOS):
**Ubuntu 16.04**
* Link to your project or a code example to reproduce issue:

**Comment by alexellis** [comment]
Environmental variables are used widely so it might be something more subtle. Also check the function logs. 

Check the service, deployment logs and gateway logs. 

Is this a local or remote cluster? What backend are you using? (Issue template asks for this info)

**Comment by ericstoekl** [comment]
Updated the original post with Environment info. This error is seen when trying to access the function locally (localhost:8080/) or remotely.

I checked the logs for `docker service logs func_gateway` and `docker service logs gif-maker` and both don't show any info. `func_gateway` logs show the function being called, but no errors. `gif-maker` only shows the `Writing lock-file to: /tmp/.lock` line, but nothing else. How do I get the deployment logs?

Is it necessary to set the `read_timeout` and `write_timeout` on the gateway in order for it to work? If so, that would make sense and there is no bug here.
